backend/tournesol/tests/test_exports.py

```diff
@@ -434,3 +445,55 @@ def test_all_export_sorts_by_username(self):
                 usernames = [row["public_username"] for row in rows]
                 self.assertEqual("a", usernames[0])
                 self.assertEqual("z", usernames[-1])
+
+    def test_all_export_comparisons_sorted_by_username_and_weekdate(self):
+        first_user = UserFactory(username="zoe")
+        last_user = UserFactory(username="alain")
+        
+        # comparisons#1, on a given Monday
+        with MockNow.Context(datetime(2023, 1, 16, tzinfo=timezone.utc)) as mock_date_1:
+            self.add_comparison(user=first_user, is_public=True)
+            self.add_comparison(user=last_user, is_public=True)
+
+        # comparisons#2, the same week on a Sunday
+        with MockNow.Context(datetime(2023, 1, 22, tzinfo=timezone.utc)) as mock_date_2:
+            self.add_comparison(user=first_user, is_public=True)
+            self.add_comparison(user=last_user, is_public=True)
+
+        # comparisons#3. random date in current week
+        random_date_this_week = datetime.now()
+        random_date_this_week = random_date_this_week - timedelta(random_date_this_week.weekday()) + timedelta(random.randint(0, 6))
```

Comment:
I think as you mock a Sunday as now it simplify a bit and you could use the function `time_ahead` that I added recently, just need to import:

```python
from core.utils.time import time_ahead
```

```suggestion
        with MockNow.Context(datetime(2023, 1, 22, tzinfo=timezone.utc)) as mock_date_2:
            self.add_comparison(user=first_user, is_public=True)
            self.add_comparison(user=last_user, is_public=True)

        # comparisons#3. random date in current week
        random_date_this_week = time_ahead(day=random.randint(0, 6))
```

In fact I am not just MockNow will work in this case...

Reply:
@aidanjungo yes I intentionally create two comparisons one on Monday and one on Sunday and make sure both are "truncated" within the same week.

Your `time_ahead` function is relevant, I'll give it a try!